File: customcads/src/interfaces/cad.ts
import ICategory from "./category"
import ICoordinates from "./coordinates"

export default interface ICad {
    id: number
    name: string
    description: string
    creatorName: string
    creationDate: string
    price: number
    cadPath: string
    imagePath: string
    camCoordinates: ICoordinates
    panCoordinates: ICoordinates
    status: string
    ordersCount: number
    category: ICategory
}

export const emptyCad: ICad = {
    id: 0,
    name: '',
    description: '',
    creatorName: '',
    creationDate: '',
    price: 0,
    cadPath: '',
    imagePath: '',
    camCoordinates: { x: 0, y: 0, z: 0 },
    panCoordinates: { x: 0, y: 0, z: 0 },
    status: '',
    ordersCount: 0,
    category: { id: 0, name: '' },
}

File: CustomCADs.Domain/Cads/Cad.cs
using CustomCADs.Domain.Cads.Enums;
using CustomCADs.Domain.Cads.ValueObjects;
using CustomCADs.Domain.Categories;
using CustomCADs.Domain.Users;

namespace CustomCADs.Domain.Cads;

public class Cad
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public CadStatus Status { get; set; }
    public decimal Price { get; set; }
    public DateTime CreationDate { get; set; }
    public Coordinates CamCoordinates { get; set; } = new();
    public Coordinates PanCoordinates { get; set; } = new();
    public Paths Paths { get; set; } = new();
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public required string CreatorId { get; set; }
    public User Creator { get; set; } = null!;
}
